### PR TITLE
resolves #577 normalize setting names and group by category

### LINKED ISSUE #577: Normalize setting names/keys
Currently, setting keys are inconsistent. We are using both camelCase and `_` as a separator. Given that the key is used to construct a title, I think we should use a strict convention.

#### Official documentation

> The properties in your configuration will be a dictionary of configuration properties.
> In the settings UI, your configuration key will be used to namespace and construct a title. Though an extension can contain multiple categories of settings, each setting of the extension must still have its own unique key. Capital letters in your key are used to indicate word breaks. For example, if your key is gitMagic.blame.dateFormat, the generated title for the setting will look like this:

> Blame: Date Format

> Entries will be grouped according to the hierarchy established in your keys. So for example, these entries
> ```
> gitMagic.blame.dateFormat
> gitMagic.blame.format
> gitMagic.blame.heatMap.enabled
> gitMagic.blame.heatMap.location
> ```
> will appear in a single group like this:

> Blame: Date Format
> Blame: Format
> Blame › Heatmap: Enabled
> Blame › Heatmap: Location

https://code.visualstudio.com/api/references/contribution-points#contributes.configuration

#### Present state

- `asciidoc.asciidoctorpdf_command`				
- `asciidoc.previewFrontMatter`
- `asciidoc.preview.style`
- `asciidoc.preview.attributes`
- `asciidoc.preview.breaks`
- `asciidoc.preview.linkify`
- `asciidoc.preview.fontFamily`
- `asciidoc.preview.fontSize`
- `asciidoc.preview.lineHeight`
- `asciidoc.preview.useEditorStyle`
- `asciidoc.preview.refreshInterval`
- `asciidoc.preview.scrollPreviewWithEditor`
- `asciidoc.preview.scrollPreviewWithEditorSelection`
- `asciidoc.preview.markEditorSelection`
- `asciidoc.preview.scrollEditorWithPreview`
- `asciidoc.preview.doubleClickToSwitchToEditor`
- `asciidoc.preview.openAsciiDocLinks`
- `asciidoc.trace`
- `asciidoc.use_asciidoctorpdf`
- `asciidoc.use_kroki`
- `asciidoc.registerAsciidoctorExtensions`
- `asciidoc.wkhtmltopdf_path`
- `asciidoc.forceUnixStyleSeparator`
- `asciidoc.enableErrorDiagnostics`
- `asciidoc.useWorkspaceRoot`

<details>
    <summary>Settings in VS code</summary>
    

</details>

##### Removed

- `asciidoc.previewFrontMatter` -> I think we should remove this setting since it's possible to configure `skip-front-matter` as an attribute: https://docs.asciidoctor.org/asciidoctor/latest/html-backend/skip-front-matter/
- `asciidoc.preview.scrollPreviewWithEditorSelection` -> Replaced by `asciidoc.preview.scrollPreviewWithEditor` (was deprecated)
- `asciidoc.forceUnixStyleSeparator` -> We are not using it??

##### Renamed

- `asciidoc.asciidoctorpdf_command` -> `asciidoc.pdf.asciidoctorPdfCommand`: Asciidoc > Pdf > Asciidoctor Pdf Command
- `asciidoc.wkhtmltopdf_path` -> `asciidoc.pdf.wkhtmltopdfCommand`: Asciidoc > Pdf > Wkhtmltopdf Command
   - Note: to be consistent with `asciidoc.pdf.asciidoctorPdfCommand`, we should support command line arguments, for instance: `/path/to/wkhtmltopdf --enable-local-file-access`
- `asciidoc.use_asciidoctorpdf` -> `asciidoc.pdf.engine`: Asciidoc > Pdf > Engine
   - Either: `wkhtmltopdf` or `asciidoctor-pdf` (it also makes room for `asciidoctor-web-pdf` if we want to enable experimental support)
- `asciidoc.preview.attributes` -> `asciidoc.preview.asciidoctorAttributes`: Asciidoc > Preview > Asciidoctor Attributes
- `asciidoc.preview.openLinksToAsciidocFiles`: Asciidoc > Preview > Open Links To Asciidoc Files
   - Note: Asciidoc is not properly capitalized but it would be even worth to use `AsciiDoc` since it will produce "Ascii Doc"
- `asciidoc.use_kroki` -> `asciidoc.extensions.enableKroki`: Asciidoc > Extensions > Enable Kroki
- `asciidoc.registerAsciidoctorExtensions` -> `asciidoc.extensions.registerWorkspaceExtensions`: Asciidoc > Extensions > Register Workspace Extensions
- `asciidoc.useWorkspaceRoot` -> `asciidoc.useWorkspaceRootAsBaseDirectory`: Asciidoc > Use Workspace Root As Base Directory

### PR DESCRIPTION
**Pull request by Mogztter**
- use third-person singular verb form
- introduce 5 categories : preview, pdf, extensions, general, debug
- use order to move most commonly used settings on top (of each category)
- deprecate/rename settings to comply with the new naming convention
- remove unused settings
- update English localization
- use `markdownDeprecationMessage` and `markdownDescription` to format deprecation message and description (using Markdown)
- add `scope: resource` (taken form the Markdown extension)

resolves #577

### TODO

- [x]  Update the Japanese localization
- [x] Add a CHANGELOG entry
- [x] Update code to use setting new names
- [x] Update the logic around the `wkhtmltopdf` command (we need to support command line arguments)
- [x] Update the logic to select a PDF engine
- [x] Update README
- [x] install `asciidoctor-pdf` in the global storage: "A global storage path pointing to a local directory where your extension has read/write access. This is a good option if you need to store large files that are accessible from all workspaces." https://code.visualstudio.com/api/references/vscode-api#ExtensionContext.globalStoragePath
- [x] try to use `asciidoctor-pdf` first and then `bundle exec asciidoctor-pdf` if the `asciidoctor-pdf` command has not been defined (i.e., default value).
    - if the user is using a custom value we should not offer to install `asciidoctor-pdf`

### CODE REVIEW

**Comment by Mogztter** [comment]
### Settings

asciidoc-settings

### Deprecation message

deprecated-kroki

**Comment by Mogztter** [comment]
@YoshihideShirai would you be able to update the Japanese localization? sorry about that 🙇🏻

**Comment by YoshihideShirai** [comment]
OK, I will update it.

File: i18n/jpn/src/asciidoctorWebViewConverter.i18n.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (Mogztter) [inline i18n/jpn/src/asciidoctorWebViewConverter.i18n.json:1]: @YoshihideShirai I'm using English as placeholder, feel free to update these files

**Pull request merged**